js/index.js

```diff
@@ -1,42 +1,87 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+//Get price and quantity values and convert them to numbers
+
+  const price = product.querySelector('.price span').innerHTML;
+  const quantity = product.querySelector('.quantity input').value;
+  console.log(Number(price), Number(quantity));
+
+// Get the subtotal value
+// Render the subtotal value in element class ".subtotal"
+
+  const subtotal = Number(price) * Number(quantity);
+
+  const subtotalElement = product.querySelector('.subtotal span');
+  subtotalElement.innerHTML = subtotal;
+  return subtotal;
 
-  //... your code goes here
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
+  
   // ITERATION 2
-  //... your code goes here
+  // Add new product to html
+  // call new function uptate total
 
+  const cartItemElements = document.getElementsByClassName('cart-item');
+
+  let totalValue = 0;
+
+  for (cartItemElement of cartItemElements) {
+    totalValue += updateSubtotal(cartItemElement);
+  }
+  
+  totalValue;
   // ITERATION 3
-  //... your code goes here
+
+  const totalValueSpan = document.querySelector('#total-value span').innerHTML = totalValue;
```

Comment:
Line 35 is unnecessary (totalValue has been defined and modified via line 29-33)

Since you're not looking to use the stored value then you don't need to store it in a variable. You can just write  `document.querySelector('#total-value span').innerHTML = totalValue;`